osf/models/mixins.py

```diff
@@ -1990,9 +1990,10 @@ def confirm_ham(self, save=False):
         """
         super().confirm_ham()
 
-        if self.logs.filter(action__in=[self.log_class.FLAG_SPAM, self.log_class.CONFIRM_SPAM]):
+        if self.logs.filter(action=self.log_class.FLAG_SPAM):
+            flag_spam_log = self.logs.filter(action=self.log_class.FLAG_SPAM).latest()
             # ensures only 'accepted' status preprints/any nodes get made public
-            if self.type == 'osf.node' or (self.type == 'osf.preprint' and self.machine_state == DefaultStates.ACCEPTED.value):
+            if flag_spam_log.params['was_public'] and self.type == 'osf.node' or (self.type == 'osf.preprint' and self.machine_state == DefaultStates.ACCEPTED.value):
```

Comment:
There

Since there'll be a bunch of preprints that are already marked as spam, we'll need to be able to handle that case. Maybe something like:

```suggestion
            if flag_spam_log.params.get('was_public', False) and self.type == 'osf.node' or (self.type == 'osf.preprint' and self.machine_state == DefaultStates.ACCEPTED.value):
```